build_ree.sh

```diff
@@ -2,45 +2,59 @@
 
 set -e
 
-FULL_VERSION="1.8.7-2012.02"
-FULL_NAME="ruby-enterprise-${FULL_VERSION}"
+function init() {
+  FULL_VERSION="1.8.7-2012.02"
+  FULL_NAME="ruby-enterprise-${FULL_VERSION}"
+  TEMP_DIR=$(mktemp -d)
 
-TEMP_DIR=$(mktemp -d)
+  echo "Serving from /tmp"
 
-cd $TEMP_DIR
+  cd /tmp
 
-rm -rf /tmp/tmp.*
+  python -m SimpleHTTPServer $PORT &
+}
 
-echo "Serving from ${TEMP_DIR}"
-cd /tmp
-python -m SimpleHTTPServer $PORT &
+function download_and_patch() {
+  init
 
-curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $TEMP_DIR
+  curl https://rubyenterpriseedition.googlecode.com/files/${FULL_NAME}.tar.gz -s -o - | tar zxf -
+  cd $FULL_NAME/source
+  curl -o 34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/34ba44f94a62c63ddf02a045b6f4edcd6eab4989.patch
+  curl -o 5384967a015be227e16af7a332a50d45e14ed0ad.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/5384967a015be227e16af7a332a50d45e14ed0ad.patch
+  curl -o tcmalloc_declare_memalign_volatile.patch https://raw.githubusercontent.com/Genius/ree-1.8.7-heroku/master/patches/tcmalloc_declare_memalign_volatile.patch
```

Comment:
Hmm, still seems weird to curl these when the files are available in the local filesystem of this repo?